tsconfig.json

```diff
@@ -7,7 +7,8 @@
     "skipLibCheck": true,
     "module": "commonjs",
     "target": "es6",
-    "jsx": "react",
+    "jsx": "react-jsx",
+    "jsxImportSource": "@emotion/react",
     "resolveJsonModule": true
```

Comment:
This tsconfig is making me itchy. Once this is merged, I'd love to turn on `strict: true` and go over the whole repo with a broom 😅
Any objections?

Reply:
@phryneas 100%. I was also bothered by this. So many places where types are just plain missing (especially component props), so I'm not opposed at all. I avoided doing it in this PR since it was already getting a bit lengthy with all the upgrades to the packages, but I'd love for that improvement.